### PR TITLE
RBAC implemented.

### PR DESCRIPTION
**Pull request by AnvarjonShavqiyev**
- Added roles to the User schema.
- Implemented role-based access control in the backend and frontend.
- Modified the React app to handle different user roles and restrict access to certain features.

photo_2024-07-06_16-56-56
photo_2024-07-06_16-57-19

### CODE REVIEW

File: my-react-app/src/components/navbar/Navbar.css [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+.logout-btn {
+  background-color: #fff;
+  border: none;
+  padding: 8px 18px;
+  font-size: 18px;
+  border-radius: 18px;
+  cursor: pointer;
+}/*# sourceMappingURL=Navbar.css.map */
```
Comment (VladKetris) [inline my-react-app/src/components/navbar/Navbar.css:20]: what this comment is for?

File: my-react-app/src/components/users-table/UserTable.tsx [hunk 2]
```diff
@@ -108,14 +110,14 @@ const UserTable: React.FC = () => {
             </Popconfirm>
           </span>
         ) : (
-          <div className="actions__wrapper">
+          userRole == 'admin' ? <div className="actions__wrapper">
```
Comment (VladKetris) [inline my-react-app/src/components/users-table/UserTable.tsx:113]: if there are several roles please create object with this roles and use it here like "userRole == USER_ROLES.ADMIN"

File: my-react-app/src/helpers/index.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const validateToken = (token:string) => {
+    const payload = token.split(".")[1];
+    if (payload) {
+      try {
+        const unHashedToken = JSON.parse(atob(payload))
+        const exp = unHashedToken.exp;
```
Comment (VladKetris) [inline my-react-app/src/helpers/index.ts:6]: nitpick: please use more meaningful naming for constants in future

File: my-react-app/src/helpers/index.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        localStorage.setItem("role", unHashedToken.role)
+        const now = new Date().getTime() / 1000;
```
Comment (VladKetris) [inline my-react-app/src/helpers/index.ts:8]: same

File: my-react-app/src/pages/Auth/Auth.css [hunk 5]
```diff
@@ -0,0 +1,68 @@
+.auth-wrapper {
+  display: flex;
```
Comment (VladKetris) [inline my-react-app/src/pages/Auth/Auth.css:2]: you have a few selectors with the same styles here, please connect it in one selector to keep it DRY:
.auth-wrapper, .auth-form-wrapper {
display: flex;
flex-direction: column;
align-items: center;
}

File: my-react-app/src/pages/Auth/Auth.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  position: absolute;
+  bottom: -10px;
+  left: 0;
+  content: "";
+  height: 1.5px;
+  width: 200px;
+  background-color: #007df9;
+}
+
+.auth-wrapper {
```
Comment (VladKetris) [inline my-react-app/src/pages/Auth/Auth.css:55]: why this selector repeats ?

File: my-react-app/src/pages/Auth/signin/Signin.tsx [hunk 7]
```diff
@@ -2,13 +2,13 @@ import { Form, Input, Button } from 'antd';
...
-import { signInUser } from '../../../redux/features/AuthSlice';
-import { User } from '../../../types';
+import { signUpUser } from '../../../redux/features/AuthSlice';
 
 const Signin = () => {
   const dispatch = useDispatch<AppDispatch>()
-  const onFinish = (values:any) => {
-    dispatch(signInUser({ email: values.email }));
+  const onFinish = (values: { email: string; password: string }) => {
+    console.log('Signin values:', values);
```
Comment (VladKetris) [inline my-react-app/src/pages/Auth/signin/Signin.tsx:10]: do you still need it here ?

File: my-react-app/src/components/users-table/UserTable.tsx (hunk 2 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/components/users-table/UserTable.tsx:113]: I understood

File: my-react-app/src/helpers/index.ts (hunk 3 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/helpers/index.ts:6]: I understood

File: my-react-app/src/helpers/index.ts (hunk 4 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/helpers/index.ts:8]: I understood

File: my-react-app/src/pages/Auth/Auth.css (hunk 5 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/pages/Auth/Auth.css:2]: Yeah, that is true.I use selectors like this in future

File: my-react-app/src/pages/Auth/Auth.css (hunk 6 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/pages/Auth/Auth.css:51]: fixed

File: my-react-app/src/pages/Auth/signin/Signin.tsx (hunk 7 above)
Comment (AnvarjonShavqiyev) [reply my-react-app/src/pages/Auth/signin/Signin.tsx]: fixed

**Comment by VladKetris** [comment]
You don't need to create separate commit for each file, you could do all the fixes in one

**Review by VladKetris — APPROVED** [review]

**Pull request merged**